app/repository/corpus.py

```diff
@@ -0,0 +1,108 @@
+from typing import Any, Mapping
+
+from db_client.models.dfce.family import Corpus, Family, FamilyCategory, FamilyCorpus
+from db_client.models.organisation import CorpusType, Organisation
+from sqlalchemy import func
+from sqlalchemy.orm import Session
+
+from app import config
+from app.models.config import CorpusConfig, CorpusTypeConfig
+
+
+def _get_family_stats_per_corpus(db: Session, corpus_import_id: str) -> dict[str, Any]:
+    total = (
+        db.query(Family)
+        .join(FamilyCorpus, FamilyCorpus.family_import_id == Family.import_id)
+        .filter(FamilyCorpus.corpus_import_id == corpus_import_id)
+        .count()
+    )
+
+    counts = (
+        db.query(Family.family_category, func.count())
+        .join(FamilyCorpus, FamilyCorpus.family_import_id == Family.import_id)
+        .filter(FamilyCorpus.corpus_import_id == corpus_import_id)
+        .group_by(Family.family_category)
+        .all()
+    )
+    found_categories = {c[0].value: c[1] for c in counts}
+    count_by_category = {}
+
+    # Supply zeros when there aren't any
```

Comment:
Could we argue that this function could in fact be two repository layer functions (one for retrieving total count, one for retrieving category count), and one service layer function (the business logic for creating the family stats per corpus object)?

Reply:
The repo layer should only really be for CRUDding for an entity/value object, whereas the service layer is for the application logic in DDD (I believe!).